### PR TITLE
[12.0][MIG] web_ir_actions_act_multi

### PR DESCRIPTION
**Pull request by alexey-pelykh**

### CODE REVIEW

**Comment by alexey-pelykh** [comment]
@pnajman-modoolar @simahawk @lreficent since you were involved in 11.0, please review this migration for 12.0

**Review by rodrigets — APPROVED** [review]

**Review by tarteo — CHANGES_REQUESTED** [review]
Can you change the name of the module to web_ir_action_act_multi (singular) and take a look at the jslint issues?

**Comment by alexey-pelykh** [comment]
@tarteo re: renaming yet there are other `web_ir_actions_*` modules - won't this break consistency? Maybe we should postpone it until 13.0 migration?

**Comment by hbrunn** [comment]
I know the conventions say no plurals, but I think we should apply this rule with some context. I started naming those modules like that years ago because that's how the models in Odoo are called. In my opninion, we shouldn't change this.

**Comment by pedrobaeza** [comment]
Yes, on guidelines it's said that names that are plurals in Odoo, should be preserved: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#11modules, so it's OK to leave it as is.

**Review by tarteo — APPROVED** [review]
:+1: Thanks. code looks good. I couldn't test the functionality though. Maybe you can add demo action so it's easier to test the functionality.

**Comment by alexey-pelykh** [comment]
@tarteo thanks for a hint, I'll consider adding example as a separate module. Not sure if this module is used a lot elsewhere, yet at least it's being used in this one https://github.com/OCA/hr-timesheet/pull/139

**Comment by hbrunn** [comment]
you can just add some example action as demo data, much less work and we don't clutter out repos with a lot of _demo modules.

**Comment by alexey-pelykh** [comment]
@hbrunn a snippet or a link to example would be much appreciated, since I haven't yet worked with demo data.

File: web_ir_actions_act_multi/static/src/js/web_ir_actions_act_multi.js [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
-"use strict";
+    "use strict";
 
     var ActionManager = require('web.ActionManager');
 
     ActionManager.include({
 
-        execute_ir_actions_act_multi: function(actions, options, index){
+        /**
+         * Intersept action handling to detect extra action type
```
Comment (hbrunn) [inline web_ir_actions_act_multi/static/src/js/web_ir_actions_act_multi.js]: interCept

**Review by hbrunn — APPROVED** [review]

File: web_ir_actions_act_multi/static/src/js/web_ir_actions_act_multi.js (hunk 1 above)
Comment (alexey-pelykh) [reply web_ir_actions_act_multi/static/src/js/web_ir_actions_act_multi.js]: Thanks!

**Comment by hbrunn** [comment]
check out https://github.com/OCA/web/blob/10.0/web_ir_actions_act_window_page/__manifest__.py#L18

**Comment by hbrunn** [comment]
whatever you add to the `demo` key in the manifest is only loaded when the demo flag for the module is set, which is never the case on production servers (if not somebody messed up majorly).

Meant to add data needed for tests, but we can use it perfectly fine to, well, demo a module's functionality.

**Comment by alexey-pelykh** [comment]
@hbrunn thought about examples - this is so purely developer-related module, not even sure it's worth adding an example besides describing what this module does. If you have any particular ideas for demos - it would be great to hear them!

**Comment by hbrunn** [comment]
hmm, I just answered your question on how to do @tarteo's proposal, don't have much of an opinion about this. As everyone approved, I'll just merge.

**Comment by alexey-pelykh** [comment]
@hbrunn true, sorry for mis-tagging!

**Pull request merged**